app/services/katello/pulp3/content_view_version/import.rb

```diff
@@ -45,14 +45,67 @@ def destroy_importer(importer_href)
         end
 
         class << self
-          def check_permissions!(path)
+          def check!(content_view:, metadata:, path:)
+            check_permissions!(metadata: metadata, path: path)
+            check_metadata!(content_view: content_view, metadata: metadata)
+          end
+
+          # rubocop:disable Metrics/MethodLength
+          # rubocop:disable Metrics/AbcSize
+          def check_metadata!(content_view:, metadata:)
+            major = metadata[:content_view_version][:major]
+            minor = metadata[:content_view_version][:minor]
+
+            if ::Katello::ContentViewVersion.where(major: major, minor: minor, content_view: content_view).exists?
+              fail _("Content View Version specified in the metadata - '%{name}' already exists. "\
+                      "If you wish to replace the existing version, delete %{name} and try again. " % { name: "#{content_view.name} #{major}.#{minor}" })
+            end
+
+            if metadata[:from_content_view_version].present?
+              from_major = metadata[:from_content_view_version][:major]
+              from_minor = metadata[:from_content_view_version][:minor]
+
+              unless ::Katello::ContentViewVersion.where(major: from_major, minor: from_minor, content_view: content_view).exists?
+                fail _("Prior Content View Version specified in the metadata - '%{name}' does not exist. "\
+                        "Please import the metadata for '%{name}' before importing '%{current}' " % { name: "#{content_view.name} #{from_major}.#{from_minor}",
+                                                                                                      current: "#{content_view.name} #{major}.#{minor}"})
+              end
+            end
+
+            repos_from_data = metadata[:repository_mapping].values.map do |value|
+              ::Katello::Repository.joins(:root, :product).where("#{::Katello::Product.table_name}" => {:name => value[:product]},
+                                                                 "#{::Katello::RootRepository.table_name}" => {:name => value[:repository]})
+            end
+
+            matching_repos = content_view.repositories.where("#{::Katello::Repository.table_name}.id" => repos_from_data.inject(&:or))
```

Comment:
What does `.inject(&:or)` do?  Is there a clearer way to write whatever this is?

Reply:
So in Active record its a little complicated to  do or's -> for example  `where (foo = 10 and bar =100)  or (foo = 20 and bar=200) or (foo = 20 and bar=300)` 
The way you do it in active record is you individually create and array  that looks like 
 `[ Repository.where(foo:100, bar:100), Repository.where(foo:20, bar:200),  Repository.where(foo:20, bar:300) ]` and then combine em with a `.inject(&:or)`